framework/math/PETScUtils/petsc_utils_04_vecmanip.cc

```diff
@@ -2,54 +2,72 @@
 
 #include "chi_log.h"
 
-//###################################################################
+// ###################################################################
 /**Copies a PETSc vector to a STL vector. Only the local portion is
  * copied.*/
-void chi_math::PETScUtils::CopyVecToSTLvector(
-  Vec x, std::vector<double>& data, size_t N)
+void chi_math::PETScUtils::CopyVecToSTLvector(Vec x,
+                                              std::vector<double>& data,
+                                              size_t N,
+                                              bool resize_STL /*=true*/)
 {
-  data.clear();
-  data.resize(N,0.0);
+  if (resize_STL)
```

Comment:
If resize_STL is false, we should either assert(data.size() == N) or throw an error if data.size() != N.

Reply:
Good recommendation. I would ammend it to data.size() >= N because you could pass a part of ghosted vector.